File: JULY-2025/30-07-2025/p2419.ts
function longestSubarray(nums: number[]): number {
  const max = Math.max(...nums)
  let maxSubArrLength = 0
  let count = 0

  for (let i = 0; i < nums.length; i++) {
    if(nums[i] === max){
      count++
    }else if(count !== 0){
      maxSubArrLength = Math.max(maxSubArrLength, count)
      count = 0
    }
  }

  return  Math.max(maxSubArrLength, count)
};

File: JULY-2025/30-07-2025/p2419.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int n = nums.size();
        int maxNums = nums[0];
        int count = 1;
        int ans = 1;
        for (int i = 1; i < n; i++) {
            if (nums[i] > maxNums) {
                maxNums = nums[i];
                count = 1;
                ans = count;
            } 
            else if (nums[i] == maxNums) {
                count++;
            } 
            else {
                // Some small number
                count = 0;
            }
            ans = max(ans, count);
        }
        return ans;
    }
};